src/pfcp/3gpp_29.244.hpp

```diff
@@ -602,6 +602,67 @@ class pfcp_cause_ie : public pfcp_ie {
   }
 };
 //-------------------------------------
+// IE ENTERPRISE SPECIFIC
+class pfcp_enterprise_specific_ie : public pfcp_ie {
+ public:
+  uint16_t enterprise_id;
+  std::string proprietary_data;
+
+  //--------
+  explicit pfcp_enterprise_specific_ie(const pfcp::enterprise_specific_t& b)
+      : pfcp_ie(PFCP_IE_ENTERPRISE_SPECIFIC) {
+    enterprise_id    = b.enterprise_id;
+    proprietary_data = b.proprietary_data;
+    tlv.set_length(2 + proprietary_data.size());
+  }
+  //--------
+  pfcp_enterprise_specific_ie() : pfcp_ie(PFCP_IE_ENTERPRISE_SPECIFIC) {
+    enterprise_id    = 0;
+    proprietary_data = {};
+    tlv.set_length(2);
+  }
+  //  --------
+  explicit pfcp_enterprise_specific_ie(const pfcp_tlv& t)
+      : pfcp_ie(t), enterprise_id(0), proprietary_data(){};
+
+  //--------
+  void to_core_type(pfcp::enterprise_specific_t& b) {
+    b.enterprise_id    = enterprise_id;
+    b.proprietary_data = proprietary_data;
+  }
+  //--------
+  void dump_to(std::ostream& os) {
+    tlv.dump_to(os);
+    os.write(
+        reinterpret_cast<const char*>(&enterprise_id), sizeof(enterprise_id));
+    os << enterprise_id;
+  }
```

Comment:
no serialization of proprietary_data ?

Reply:
Why do write twice enterprise_id and proprietary_data ?
seems to me suspicious reinterpret_cast<const char*>(&proprietary_data) ...